specification/communication/data-plane/Messages/stable/2024-02-01/communicationservicesmessages.json

```diff
@@ -0,0 +1,1051 @@
+        "values": [
+          {
+            "name": "quickReply",
+            "value": "quickReply",
+            "description": "The WhatsApp button sub type is quick reply."
+          },
+          {
+            "name": "url",
+            "value": "url",
+            "description": "The WhatsApp button sub type is url."
+          }
+        ]
+      }
+    },
+    "WhatsAppMessageTemplateBindings": {
+      "type": "object",
+      "description": "The template bindings for WhatsApp",
+      "properties": {
+        "header": {
+          "type": "array",
+          "description": "The header template bindings",
+          "items": {
+            "$ref": "#/definitions/WhatsAppMessageTemplateBindingsComponent"
+          },
+          "x-ms-identifiers": []
+        },
+        "body": {
+          "type": "array",
+          "description": "The body template bindings",
+          "items": {
+            "$ref": "#/definitions/WhatsAppMessageTemplateBindingsComponent"
+          },
+          "x-ms-identifiers": []
+        },
+        "footer": {
+          "type": "array",
+          "description": "The footer template bindings",
+          "items": {
+            "$ref": "#/definitions/WhatsAppMessageTemplateBindingsComponent"
+          },
+          "x-ms-identifiers": []
+        },
+        "buttons": {
+          "type": "array",
+          "description": "The button template bindings",
+          "items": {
+            "$ref": "#/definitions/WhatsAppMessageTemplateBindingsButton"
+          },
+          "x-ms-identifiers": []
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/MessageTemplateBindings"
+        }
+      ],
+      "x-ms-discriminator-value": "whatsApp"
+    },
+    "WhatsAppMessageTemplateBindingsButton": {
+      "type": "object",
+      "description": "The template bindings component button for WhatsApp",
+      "properties": {
+        "subType": {
+          "$ref": "#/definitions/WhatsAppMessageButtonSubType",
+          "description": "The WhatsApp button sub type"
+        },
+        "refValue": {
+          "type": "string",
+          "description": "The name of the referenced item in the template values."
+        }
+      },
+      "required": [
+        "refValue"
+      ]
+    },
+    "WhatsAppMessageTemplateBindingsComponent": {
+      "type": "object",
+      "description": "The template bindings component for WhatsApp",
+      "properties": {
+        "refValue": {
+          "type": "string",
+          "description": "The name of the referenced item in the template values."
+        }
+      },
+      "required": [
+        "refValue"
+      ]
+    },
+    "WhatsAppMessageTemplateItem": {
+      "type": "object",
+      "description": "The WhatsApp-specific template response contract",
+      "properties": {
+        "content": {
+          "description": "WhatsApp platform's template content\nThis is the payload returned from WhatsApp\nAPI."
```

Comment:
Recommend removing the multiple "\n"s in this description.

Reply:
removed by removing multiple-line docs from TypeSpec